src/modules/position-creator/services/position.creator.transaction.ts

```diff
@@ -173,64 +175,65 @@ export class PositionCreatorTransactionService {
-        if (lockEpochs) {
-            endpointArgs = [
-                new U64Value(new BigNumber(lockEpochs)),
-                new BigUIntValue(amount0Min),
-                new BigUIntValue(amount1Min),
-                ...swapRouteArgs,
-            ];
-        } else {
-            endpointArgs = [
-                new AddressValue(Address.fromBech32(farmAddress)),
-                new BigUIntValue(amount0Min),
-                new BigUIntValue(amount1Min),
-                ...swapRouteArgs,
-            ];
-        }
-
         const gasLimit =
             gasConfig.positionCreator.singleToken.farmPosition +
             gasConfig.pairs.addLiquidity +
             gasConfig.farms[FarmVersion.V2].enterFarm.withTokenMerge +
             gasConfig.pairs.swapTokensFixedInput.withFeeSwap *
                 swapRoutes[0].pairs.length;
 
-        let interaction = contract.methodsExplicit
-            .createFarmPosFromSingleToken(endpointArgs)
-            .withSender(Address.fromBech32(sender))
-            .withChainID(mxConfig.chainID)
-            .withGasLimit(gasLimit);
+        const transactionOptions = new TransactionOptions({
+            sender: sender,
+            chainID: mxConfig.chainID,
+            gasLimit: gasLimit,
+            function: 'createFarmPosFromSingleToken',
+            arguments: [
+                new AddressValue(Address.newFromBech32(farmAddress)),
+                new BigUIntValue(amount0Min),
+                new BigUIntValue(amount1Min),
+                VariadicValue.fromItems(...swapRouteArgs),
+            ],
+        });
 
         if (
             payments[0].tokenIdentifier === mxConfig.EGLDIdentifier &&
             payments.length === 1
         ) {
-            interaction = interaction.withValue(
-                new BigNumber(payments[0].amount),
-            );
+            transactionOptions.nativeTransferAmount = payments[0].amount;
         } else {
             payments[0].tokenIdentifier =
                 payments[0].tokenIdentifier === mxConfig.EGLDIdentifier
                     ? wrappedEgldTokenID
                     : payments[0].tokenIdentifier;
-            interaction = interaction.withMultiESDTNFTTransfer(
-                payments.map((payment) =>
-                    TokenTransfer.metaEsdtFromBigInteger(
-                        payment.tokenIdentifier,
-                        payment.tokenNonce,
-                        new BigNumber(payment.amount),
-                    ),
+            transactionOptions.tokenTransfers = payments.map(
+                (payment) =>
+                    new TokenTransfer({
+                        token: new Token({
+                            identifier: payment.tokenIdentifier,
+                            nonce: BigInt(payment.tokenNonce),
+                        }),
+                        amount: BigInt(payment.amount),
+                    }),
+            );
+        }
+
+        if (lockEpochs) {
```

Comment:
same here: i would suggest extracting 2 functions, for a better code readability

Reply:
done